packages/cells/src/widget.ts

```diff
@@ -576,8 +582,19 @@ export class Cell<T extends ICellModel = ICellModel> extends Widget {
     inputWrapper.addWidget(input);
     (this.layout as PanelLayout).addWidget(inputWrapper);
 
-    this._inputPlaceholder = new InputPlaceholder(() => {
-      this.inputHidden = !this.inputHidden;
+    this._inputPlaceholder = new InputPlaceholder({
+      callback: () => {
+        this.inputHidden = !this.inputHidden;
+      },
+      text: input.model.sharedModel.getSource().split('\n')[0]
+    });
+
+    input.model.contentChanged.connect((sender, args) => {
```

Comment:
Could you use a protected `onContentChanged` method instead of a lambda? And as it is defined in `MarkdownCell` make the necessary `super` call there.